lib/analysis/author/community.py

```diff
@@ -0,0 +1,265 @@
+# import plotly
+# from matplotlib import pyplot as plt
+# from plotly.tools import FigureFactory as FF
+# from scipy.cluster.hierarchy import dendrogram, linkage
+
+from analysis.author import ranking
+from util.read import *
+
+sys.setrecursionlimit(10000)
+
+
+def write_matrix(json_data, tree_filename="infomap/output/" + "author_graph.tree"):
+    """
+
+    :param json_data:
+    :return:
+    """
+    top_authors = set()
+    top_authors_data = dict()
+    author_scores = ranking.get(active_score=2, passive_score=1, write_to_file=False)
+    index = 0
+    for email_addr, author_score in author_scores:
+        index += 1
+        top_authors.add(email_addr)
+        top_authors_data[email_addr] = [author_score]
+        if index == 100:
+            break
+
+    print("Adding nodes to author's graph...")
+    author_graph = nx.DiGraph()
+    for msg_id, message in json_data.items():
+        if message['From'] in top_authors:
+            if message['Cc'] is None:
+                addr_list = message['To']
+            else:
+                addr_list = message['To'] | message['Cc']
+            for to_address in addr_list:
+                if to_address in top_authors:
+                    if author_graph.has_edge(message['From'], to_address):
+                        author_graph[message['From']][to_address]['weight'] *= \
+                            author_graph[message['From']][to_address]['weight'] / (author_graph[message['From']][to_address]['weight'] + 1)
+                    else:
+                        author_graph.add_edge(message['From'], to_address, weight=1)
+
+    author_graph_undirected = author_graph.to_undirected()
+    clustering_coeff = nx.clustering(author_graph_undirected)
+    in_degree_dict = author_graph.in_degree(nbunch=author_graph.nodes_iter())
+    out_degree_dict = author_graph.out_degree(nbunch=author_graph.nodes_iter())
+
+    for email_addr in top_authors:
+        top_authors_data[email_addr].append(in_degree_dict[email_addr])
+        top_authors_data[email_addr].append(out_degree_dict[email_addr])
+        top_authors_data[email_addr].append(clustering_coeff[email_addr])
+
+    print("Parsing", tree_filename + "...")
+    with open(tree_filename, 'r') as tree_file:
+        for line in tree_file:
+            if not line or line[0] == '#':
+                continue
+            line = line.split()
+            if line[2][1:-1] in top_authors:
+                top_authors_data[line[2][1:-1]].append(float(line[1]))
+        tree_file.close()
+
+    with open("top_authors_data.csv", 'w') as output_file:
+        output_file.write("Email Address,Author Score,In-Degree,Out-Degree,Clustering Coeff,Module Flow\n")
+        for email_addr, data_list in top_authors_data.items():
+            output_file.write(email_addr+","+",".join([str(x) for x in data_list])+"\n")
+        output_file.close()
+    print("Authors data written to file.")
+
+
+def write_pajek(author_graph, filename="author_graph.net"):
```

Comment:
This function looks like a good candidate for lib/util module. Can we move it there?

Reply:
also please add the sphinx docs

Reply:
They are remnants of infomap community detection that save the communities detected in Pajek format and I felt should be a part of the community module in authors package. I am not sure if util would be the best fit as util is supported to assist other modules (like read functions)

Reply:
Please check b8193f4c7935c85e262dfae192c1b2d30abbec7d for documentation on community module